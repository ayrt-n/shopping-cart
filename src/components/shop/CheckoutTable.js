import React from 'react';
import CheckoutProductRow from './CheckoutProductRow';
import CheckoutTableTotal from './CheckoutTableTotal';

function CheckoutTable({ cart }) {
  return (
    <table className="Checkout-table">
      <thead>
        <tr>
          <th>Item</th>
          <th>Qty.</th>
          <th>Total</th>
        </tr>
      </thead>
      <tbody>
        {cart.map((product) => {
          return (
            <CheckoutProductRow product={product} key={product.id} />
          );
        })}
        <CheckoutTableTotal cart={cart} />
      </tbody>
    </table>
  );
}

export default CheckoutTable;
