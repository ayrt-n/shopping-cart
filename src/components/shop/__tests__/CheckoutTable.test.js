import React from 'react';
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import CheckoutTable from '../CheckoutTable';

// Setup mocks
jest.mock('../CheckoutProductRow', () => ({ product }) => (
  <tr>
    <td>{product.name}</td>
    <td>{product.quantity}</td>
    <td>{product.quantity * product.price}</td>
  </tr>
));

jest.mock('../CheckoutTableTotal', () => ({ cart }) => (
  <tr>
    <th colSpan="2">Total</th>
    <td>$TOTAL</td>
  </tr>
));

// Setup props
const cart = [
  {id: 'test1', name: 'Test1', quantity: 1, price: 1},
  {id: 'test1', name: 'Test2', quantity: 1, price: 1},
  {id: 'test1', name: 'Test3', quantity: 1, price: 1},
];

describe('CheckoutTable component', () => {
  it('correctly renders checkout table given array of products', () => {
    const { container } = render(<CheckoutTable cart={cart} />);

    expect(container).toMatchSnapshot();
  });
});

