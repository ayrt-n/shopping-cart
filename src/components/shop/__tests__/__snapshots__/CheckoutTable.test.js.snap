// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutTable component correctly renders checkout table given array of products 1`] = `
<div>
  <table
    class="Checkout-table"
  >
    <thead>
      <tr>
        <th>
          Item
        </th>
        <th>
          Qty.
        </th>
        <th>
          Total
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Test1
        </td>
        <td>
          1
        </td>
        <td>
          1
        </td>
      </tr>
      <tr>
        <td>
          Test2
        </td>
        <td>
          1
        </td>
        <td>
          1
        </td>
      </tr>
      <tr>
        <td>
          Test3
        </td>
        <td>
          1
        </td>
        <td>
          1
        </td>
      </tr>
      <tr>
        <th
          colspan="2"
        >
          Total
        </th>
        <td>
          $TOTAL
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
